sycl/test/basic_tests/is_device_copyable_neg.cpp

```diff
@@ -0,0 +1,75 @@
+// RUN: not %clangxx -fsycl -fsycl-device-only -fsyntax-only \
+// RUN: %s -I %sycl_include 2>&1 | FileCheck -check-prefix=CHECK-ERR %s
+
+// This test checks if compiler reports compilation error on an attempt to pass
+// a struct with type that is not device copyable as SYCL kernel parameter.
+
+#include <CL/sycl.hpp>
+
+using namespace cl::sycl;
+
+struct A {
+  int i;
+  A(int _i) : i(_i) {}
+  A(const A &x) : i(x.i) {}
+};
+
+struct B {
+  int i;
+  ~B();
+};
+
+// Copy of B, needed to have crisper error diagnostic checks
+struct C {
+  int i;
+  ~C();
+};
+
+// Copy of B, needed to have crisper error diagnostic checks
+struct D {
+  int i;
+  ~D();
+};
+
+struct FunctorA {
+  FunctorA() {}
+  void operator()() const {
+    (void)c;
+    (void)i;
+  }
+
+  C c;
+  int i;
+};
+
+struct FunctorB : public D {
+  FunctorB() {}
+  void operator()() const {
+    (void)i;
+    (void)j;
+  }
+
+  int j;
+};
+
+void test() {
+  A IamBad(1);
+  B IamAlsoBad{0};
+  queue Q;
+  Q.single_task<class TestA>([=] {
+    int A = IamBad.i;
+    int B = IamAlsoBad.i;
+  });
+
+  FunctorA FA;
+  Q.single_task<class TestB>(FA);
+
+  FunctorB FB;
+  Q.single_task<class TestC>(FB);
+}
+
+// CHECK-ERR: static_assert failed due to requirement 'is_device_copyable<A,
+// void> CHECK-ERR: static_assert failed due to requirement
+// 'is_device_copyable<B, void> CHECK-ERR: static_assert failed due to
+// requirement 'is_device_copyable<C, void> CHECK-ERR: static_assert failed due
+// to requirement 'is_device_copyable<D, void>
```

Comment:
```suggestion
// CHECK-ERR: static_assert failed due to requirement 'is_device_copyable<A, void>
// CHECK-ERR: static_assert failed due to requirement 'is_device_copyable<B, void>
// CHECK-ERR: static_assert failed due to requirement 'is_device_copyable<C, void>
// CHECK-ERR: static_assert failed due to requirement 'is_device_copyable<D, void>
```
You should also add the line number information to the CHECK-ERR lines so it's clear where in the test file the diagnostics are issued.

Reply:
That mess was created by clang-format. Fixed it and also added source lines numbers in https://github.com/intel/llvm/pull/3905/commits/da49819bce2df28310ba8af73934aa87fcf88ce6
Each of types B, C, D is used only once; C and D are copies of B to make the diagnostic/error checks look more obvious.